src/system/group.cr

```diff
@@ -0,0 +1,158 @@
+    return group_struct.value.gr_gid if group_struct
+    nil
+  end
+
+  # Returns the group specified by the group ID.
+  # Raises `System::Group::NotFoundError` if not found.
+  def self.get(gid : Int) : Group
+    group_struct = LibC.getgrgid(Group.int_to_gid(gid))
+    return new(group_struct.value) if group_struct
+
+    raise NotFoundError.new("Group with gid '#{gid}', was not found.")
+  end
+
+  # Returns the user specified by the group ID.
+  # Returns `nil` if not found.
+  def self.get?(gid : Int) : Group?
+    group_struct = LibC.getgrgid(Group.int_to_gid(gid))
+    return new(group_struct.value) if group_struct
+    nil
+  end
+
+  # Returns the group specified by the groupname.
+  # Raises `System::Group::NotFoundError` if not found.
+  def self.get(groupname : String) : Group
+    group_struct = LibC.getgrnam(groupname.check_no_null_byte)
+    return new(group_struct.value) if group_struct
+
+    raise NotFoundError.new("Group with name '#{groupname}', was not found.")
+  end
+
+  # Returns the user specified by the username.
+  # Returns `nil` if not found.
+  def self.get?(groupname : String) : Group?
+    group_struct = LibC.getgrnam(groupname.check_no_null_byte)
+    return new(group_struct.value) if group_struct
+    nil
+  end
+
+  # :nodoc:
+  private def initialize(group : LibC::Group)
+    @name = String.new(group.gr_name)
+    @gid = group.gr_gid
+    slice = group.gr_mem.to_slice_null_terminated(Limits::NGROUPS_MAX)
+    @member_names = Array.new(slice.size) { |idx| String.new(slice[idx]) }
+  end
+
+  # Returns the groups name as a `String`.
+  getter name : String
+
+  # Returns the group ID.
+  getter gid : GID
+
+  # Returns an `Array` of the groups members names as `String`s.
+  getter member_names : Array(String)
+
+  # Yields the groups members names as `String`s.
+  def each_member_name(&block : String -> Nil) : Nil
+    @member_names.each { |member| yield(member) }
+  end
+
+  # Returns an `Array` of the groups members as `User`s.
+  def members : Array(User)
+    @member_names.map { |member| User.get(member) }
+  end
+
+  # Yields the groups members as `User`s.
+  def each_member(&block : User -> Nil) : Nil
+    each_member_name { |member| yield(User.get(member)) }
+  end
+
+  # See `Object#hash(hasher)`
+  def hash(hasher)
+    @gid.hash(hasher)
+  end
+
+  # Compares this group with *other*, returning `-1`, `0` or `+1` depending if the
+  # group ID is less, equal or greater than the *other* group ID.
+  def <=>(other : Group) : Int
+    @gid <=> other.gid
+  end
+
+  # Returns a `String` representation of the group. It's group name.
+  def to_s
+    @name
+  end
+
+  # Appends the username to the given `IO`.
+  def to_s(io : IO)
+    io << @name
+  end
+
+  # :nodoc:
+  def self.check_gid_in_bounds(gid : Int) : Nil
+    return if (gid >= 0) && (gid <= Limits::GID_MAX)
+    raise OutOfBoundsError.new("gid: '#{gid}' is out of bounds.")
+  end
+
+  # :nodoc:
+  def self.int_to_gid(gid : Int) : GID
```

Comment:
This method is unneccesary since integer types *are* automatically converted in C bindings (but only C bindings). You can just call `check_gid_in_bounds` in the calling methods.

Reply:
Fixed in next push.